hack/lib/init.sh

```diff
@@ -51,4 +51,10 @@ os::log::stacktrace::install
 # All of our Bash scripts need to have access to the
 # binaries that we build so we don't have to find
 # them before every invocation.
-os::util::environment::update_path_var
\ No newline at end of file
+os::util::environment::update_path_var
+
+if [[ -z "${TMP_ENV_SET-}" ]]; then
+	name="$( basename $0 )"
```

Comment:
`basename` takes as the second arg the suffix to strip, so to do this and not leak `name` var to the thread,

```sh
os::util::environment::setup_tmpdir_vars "$( basename "$0" ".sh" )"
```